src/dataflow/components/nodes/controls/plot-control.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import Rete from "rete";
+import "./plot-control.sass";
+import { Line } from "react-chartjs-2";
+import { ChartOptions } from "chart.js";
+import { cloneDeep } from "lodash";
+import { MAX_NODE_VALUES } from "../../dataflow-program";
+
+export class PlotControl extends Rete.Control {
+  private emitter: any;
+  private component: any;
+  private props: any;
+  private node: any;
```

Comment:
Perhaps not for now, but if we're going to continue to use Rete we should put some effort into having more useful types (and/or suggest that the maintainers do so). For instance, in JSXGraph we've gone ahead and defined our own types for some of the JSXGraph objects that we use.